backend/nodes/properties/inputs/numeric_inputs.py

```diff
@@ -3,21 +3,32 @@
 from .base_input import BaseInput
 
 
-def clampInt(value, min_value, max_value):
-    value = int(value)
+def clampNumber(
+    value: Union[float, int],
+    precision: int,
+    offset: Union[float, int],
+    step: Union[float, int],
+    min_value: Union[float, int],
+    max_value: Union[float, int],
+) -> Union[float, int]:
+    # Convert proper number type
+    if precision:
```

Comment:
Again, I think number conditions are hacky.

```suggestion
    if precision > 0:
```